admin/public/src/reducers/capabilities.jsx

```diff
@@ -0,0 +1,18 @@
+import { CAPABILITIES_REQUEST, CAPABILITIES_FETCH_SUCCESS } from '../actions/capabilities.jsx';
+
+export function capabilities(state = {}, action) {
+  switch (action.type) {
+    case CAPABILITIES_REQUEST:
+      return {
+        ...state,
+        inRequest: true,
+        error: null
+      };
+
+    case CAPABILITIES_FETCH_SUCCESS:
+      return { ...action.payload };
+
+    default:
```

Comment:
There's no real error callback. Not a huge deal, but the spinner will spin indefinitely.

Reply:
(since you don't set inRequest -> false)

Reply:
I made it intentionally that way. If the callback is never called, it should really fail hard. Otherwise callback should be called with `null` and the UI will handle everything else gracefully.